spec/app/domain/authentication/authn_k8s/inject_client_cert_spec.rb

```diff
@@ -0,0 +1,292 @@
+      .with(csr, hash_including(
+        :subject_altnames => [ spiffe_altname ]))
+      .and_return(webservice_signed_cert)
+
+    allow(webservice_signed_cert).to receive(:to_pem)
+      .and_return(webservice_signed_cert_pem)
+
+    allow(Authentication::AuthnK8s::K8sHost)
+      .to receive(:from_csr)
+      .with(hash_including(:account => account, :service_name => service_id, :csr => csr))
+      .and_return(k8s_host)
+
+    allow(Util::OpenSsl::X509::SmartCsr)
+      .to receive(:new)
+      .with(csr)
+      .and_return(smart_csr)
+
+    allow(Authentication::AuthnK8s::SpiffeId)
+      .to receive(:new)
+      .with(spiffe_id)
+      .and_return(spiffe_id)
+
+    allow(Authentication::AuthnK8s::PodRequest)
+      .to receive(:new)
+      .with(hash_including(:service_id => service_id,
+                           :k8s_host => k8s_host,
+                           :spiffe_id => spiffe_id))
+      .and_return(pod_request)
+
+    allow(Authentication::AuthnK8s::ValidatePodRequest)
+      .to receive(:new)
+      .and_return(validate_pod_request)
+
+      allow(host_annotation_2).to receive(:values)
+        .and_return({ :name => k8s_authn_container_name })
+      allow(host_annotation_2).to receive(:[])
+        .with(:value)
+        .and_return(nil)
+  end
+
+  subject(:injector) { Authentication::AuthnK8s::InjectClientCert
+    .new(dependencies: dependencies) }
+
+  context "invocation" do
+    context "when csr is checked" do
+      before :each do
+        allow(validate_pod_request)
+          .to receive(:call)
+          .with(no_args)
+          .and_return(nil)
+      end
+
+      it "throws CSRIsMissingSpiffeId if smart_csr.spiffe_id is not defined" do
+        error_type = Authentication::AuthnK8s::CSRIsMissingSpiffeId
+        missing_spiffe_id_error = "CSR must contain SPIFFE ID SAN"
+
+        allow(Util::OpenSsl::X509::SmartCsr)
+          .to receive(:new)
+          .with(csr)
+          .and_return(bad_spiffe_smart_csr)
+
+        expect { injector.(conjur_account: account,
+                           service_id: service_id,
+                           csr: csr) }.to raise_error(error_type, missing_spiffe_id_error)
+      end
+
+      it "throws CSRNamespaceMismatch when common_name does not match spiffe_id.namespace" do
+        error_type = Authentication::AuthnK8s::CSRNamespaceMismatch
+        wrong_cn_error = "Namespace in SPIFFE ID 'WrongNamespace' must match namespace implied " \
+          "by common name ('SpiffeNamespace')"
+
+        allow(Util::OpenSsl::X509::SmartCsr)
+          .to receive(:new)
+          .with(csr)
+          .and_return(bad_cn_namespace_smart_csr)
+
+        expect { injector.(conjur_account: account,
+                           service_id: service_id,
+                           csr: csr) }.to raise_error(error_type, wrong_cn_error)
+      end
+    end
+
+    it "raises validate_pod_request error when validate_pod_request fails" do
```

Comment:
The previous test cases use the error type in the descriptions. Can we do that here for consistency? Looks like it would be `RuntimeError`?